presto-main/src/main/java/io/prestosql/server/HttpRequestSessionContext.java

```diff
@@ -212,6 +214,7 @@ private static Identity buildSessionIdentity(Optional<Identity> authenticatedIde
                 .orElseGet(() -> Identity.forUser(user))
                 .withAdditionalRoles(parseRoleHeaders(headers))
                 .withAdditionalExtraCredentials(parseExtraCredentials(headers))
+                .withAdditionalGroups(groupProvider.getGroups(user))
```

Comment:
Add a test for this in `TestHttpRequestSessionContext`. Make sure it fetches the groups for the `PRESTO_USER` header and not the original authenticated identity.